crates/types/src/contract_payloads/tob/mod.rs

```diff
@@ -238,3 +241,27 @@ mod test {
         );
     }
 }
+
+fn compute_reward_checksum(start_tick: i32, start_liquidity: u128, quantities: &[u128]) -> U160 {
+    let mut hasher = Keccak::v256();
+    let mut reward_checksum = [0u8; 32];
+
+    // Process all tick reward updates
+    let mut liquidity = start_liquidity;
+    for (tick, &quantity) in (start_tick..).zip(quantities.iter()) {
+        let mut tick_hasher = Keccak::v256();
+        tick_hasher.update(&reward_checksum); // previous checksum
+        tick_hasher.update(&liquidity.to_le_bytes());
+        tick_hasher.update(&tick.to_le_bytes());
+
+        tick_hasher.finalize(&mut reward_checksum);
```

Comment:
couple things ive noticed. 
1) should be BE not LE
2) pretty sure alloy has as built in keccak builder so no need to add tiny keccak